inference-engine/src/gna_plugin/gna2_model_dump.cpp

```diff
@@ -0,0 +1,225 @@
+    }
+    if (paramCount > 1) {
+        uint32_t* biasVectorIndex = reinterpret_cast<uint32_t*>(parameters[BiasModeFCAffineParamIdx]);
+        if (biasVectorIndex != nullptr)
+            dumpFile << "\tParameter name: biasVectorIndex, value: " << *biasVectorIndex;
+    }
+}
+
+void dumpGMMParameters(std::ostream& dumpFile, void** parameters, int paramCount) {
+    if (paramCount > 0) {
+        uint32_t* maximumScore = reinterpret_cast<uint32_t*>(parameters[MaximumScoreParamIdx]);
+        dumpFile << "\tParameter name: maximumScore, value: " << *maximumScore;
+    }
+}
+
+void dumpRecurrentParameters(std::ostream& dumpFile, void** parameters, int paramCount) {
+    if (paramCount > 0) {
+        uint32_t* delay = reinterpret_cast<uint32_t*>(parameters[DelayParamIdx]);
+        dumpFile << "\tParameter name: delay, value: " << *delay;
+    }
+}
+
+typedef void (*dumpParamaters) (std::ostream&, void**, int);
+
+const std::map<Gna2OperationType, dumpParamaters> dumpParamMap = {
+    {Gna2OperationTypeConvolution, dumpConvolutionParameters},
+    {Gna2OperationTypeCopy, dumpCopyParameters},
+    {Gna2OperationTypeFullyConnectedAffine, dumpFCAffineParameters},
+    {Gna2OperationTypeGmm, dumpGMMParameters},
+    {Gna2OperationTypeRecurrent, dumpRecurrentParameters}
+};
+
+void dumpPwl(std::ostream & dumpFile, const Gna2Tensor& activation) {
+    const Gna2PwlSegment* const segments = static_cast<Gna2PwlSegment*>(activation.Data);
+    const uint32_t numberOfSegments = activation.Shape.Dimensions[0];
+
+    for (uint32_t k = 0; k < numberOfSegments; k++) {
+        auto scale = ((segments[k].xBase & 3) + 1) * 8;
+        int32_t B = segments[k].xBase & 0xfffffffc;
+
+        dumpFile << "\t\tBase value for input(B) : " << B << "\n";
+        dumpFile << "\t\tBase value for output(b) : " << segments[k].yBase << "\n";
+        dumpFile << "\t\tSegment slope (S): " << segments[k].Slope << "\n";
+        dumpFile << "\t\tShift(scale) : " << scale << "\n";
+        dumpFile << "\t\ty = ax + b:   a = " << segments[k].Slope / (2 ^ scale)
+            << ", b = " << segments[k].yBase - ((B * segments[k].Slope) / (2 ^ scale));
+        if (segments[k].Slope != 0) {
+            dumpFile << ", x0 = " << B - ((segments[k].yBase * (2 ^ scale)) / segments[k].Slope);
+        }
+        dumpFile << "\n";
+    }
+}
+
+void dumpCompoundBias(std::ostream& dumpFile, const Gna2Tensor& activation) {
+    const Gna2CompoundBias* const bias = static_cast<Gna2CompoundBias*>(activation.Data);
+
+    dumpFile << "\t\tBias: " << bias->Bias << ", multiplier: " << unsigned(bias->Multiplier) << "\n";
+}
+
+void DumpGna2Model(const Gna2Model & gnaModel, const std::string dumpFolderName, bool dumpData) {
+    std::stringstream dumpFileName;
+    uint32_t opsNo = gnaModel.NumberOfOperations;
+    auto now = std::chrono::system_clock::now();
+
+    dumpFileName << dumpFolderName << opsNo << "_layer_model_dump" << ".txt";
```

Comment:
add prefix to filename eg Gna2ModelDebugDump

Reply:
Done.